analyzers/src/SonarAnalyzer.Common/Rules/UseUnixEpochBase.cs

```diff
@@ -0,0 +1,122 @@
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, c.Node.GetLocation(), "DateTime"));
+                    }
+                    else if (type.DerivesFrom(KnownType.System_DateTimeOffset) && IsEpochCtor(c.Node, c.SemanticModel))
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, c.Node.GetLocation(), "DateTimeOffset"));
+                    }
+                },
+                Language.SyntaxKind.ObjectCreationExpressions);
+        });
+
+    protected static bool IsValueEqualsTo(TLiteralExpression literal, int value) =>
+        int.TryParse(literal.ChildTokens().First().ValueText, out var parsedValue) && parsedValue == value;
+
+    private bool IsEpochCtor(SyntaxNode node, SemanticModel model)
+    {
+        var methodSymbol = (IMethodSymbol)model.GetSymbolInfo(node).Symbol;
+        var lookup = Language.MethodParameterLookup(node, methodSymbol);
+
+        if (IsParameterExistingAndLiteralEqualTo("year", EpochYear, lookup)
+            && IsParameterExistingAndLiteralEqualTo("month", EpochMonth, lookup)
+            && IsParameterExistingAndLiteralEqualTo("day", EpochDay, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("hour", 0, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("minute", 0, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("second", 0, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("millisecond", 0, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("microsecond", 0, lookup)
+            && IsDateTimeKindNonExistingOrUtc(lookup)
+            && IsCalendarNonExistingOrGregorian(lookup)
+            && IsOffsetNonExistingOrZero(lookup))
+        {
+            return true;
+        }
+        return false;
+    }
+
+    private static bool IsParameterExistingAndLiteralEqualTo(string parameterName, int value, IMethodParameterLookup lookup) =>
+        lookup.TryGetSyntax(parameterName, out var expressions) && IsLiteralAndEqualTo(expressions[0], value);
+
+    private static bool IsParameterNonExistingOrLiteralEqualTo(string parameterName, int value, IMethodParameterLookup lookup) =>
+        !lookup.TryGetSyntax(parameterName, out var expressions) || IsLiteralAndEqualTo(expressions[0], value);
+
+    private static bool IsLiteralAndEqualTo(SyntaxNode node, int value) =>
+        node is TLiteralExpression literal && IsValueEqualsTo(literal, value);
+
+    private bool IsDateTimeKindNonExistingOrUtc(IMethodParameterLookup lookup) =>
+        !lookup.TryGetSyntax("kind", out var expressions)
+        || (expressions[0] is TMemberAccessExpression memberAccess && IsDateTimeKindUtc(memberAccess));
+
+    private bool IsCalendarNonExistingOrGregorian(IMethodParameterLookup lookup) =>
+        !lookup.TryGetSyntax("calendar", out var expressions) || IsGregorianCalendar(expressions[0]);
```

Comment:
There is no `TryGetSymbol` overload that accepts the parameter name. I think this is enough and we can allow the FNs you mentioned above

Reply:
I see. Those are edge cases, but you can still cover them if you introduce a helper method:
```cs
private static bool TryGetParameterSymbol(IMethodParameterLookup lookup, string parameterName, out IParameterSymbol symbol)
{
    symbol = null;
    return lookup.TryGetSyntax(parameterName, out var expressions) && lookup.TryGetSymbol(expressions[0], out symbol);
}
```
Then use it instead of `TryGetSyntax`.